app/code/Magento/QuoteGraphQl/Model/Cart/CreateQuoteAddressByCustomerAddress.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\QuoteGraphQl\Model\Cart;
+
+use Magento\CustomerGraphQl\Model\Customer\Address\GetCustomerAddress;
+use Magento\CustomerGraphQl\Model\Customer\GetCustomer;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthenticationException;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+use Magento\Quote\Model\Quote\Address;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+
+/**
+ * Creates a quote address based on given context, customer address ID and customer address
+ */
+class CreateQuoteAddressByCustomerAddress
+{
+    /**
+     * @var QuoteAddressFactory
+     */
+    private $quoteAddressFactory;
+
+    /**
+     * @var GetCustomer
+     */
+    private $getCustomer;
+
+    /**
+     * @var GetCustomerAddress
+     */
+    private $getCustomerAddress;
+
+    /**
+     * @param QuoteAddressFactory $quoteAddressFactory
+     * @param GetCustomer $getCustomer
+     * @param GetCustomerAddress $getCustomerAddress
+     */
+    public function __construct(
+        QuoteAddressFactory $quoteAddressFactory,
+        GetCustomer $getCustomer,
+        GetCustomerAddress $getCustomerAddress
+    ) {
+        $this->quoteAddressFactory = $quoteAddressFactory;
+        $this->getCustomer = $getCustomer;
+        $this->getCustomerAddress = $getCustomerAddress;
+    }
+
+    /**
+     * @param ContextInterface $context
+     * @param int|string|null $customerAddressId
+     * @param array|null $customerAddress
+     * @return Address
+     * @throws GraphQlAuthenticationException
```

Comment:
Usually `@throws` annotation tags are added only when exception is thrown explicitly in the function. Any other exceptions from parent methods are implicit and not annotated. "Missing `@throws` tag(s)" can be turned off in PHPStorm Inspections->PHP->PHPDoc

Reply:
Thanks for mentioning, @TomashKhamlai. I've added requested changes. Thank you!

Reply:
> Usually @throws annotation tags are added only when exception is thrown explicitly in the function
I little disagree with this statement. If I work with some API I will prefer to know which of exception could be thrown (without submerging in implementation)